core/src/main/java/io/confluent/kafka/schemaregistry/storage/KafkaSchemaRegistry.java

```diff
@@ -725,7 +725,7 @@ public Schema register(String subject,
       Config config = getConfigInScope(subject);
       Mode mode = getModeInScope(subject);
 
-      if (mode != Mode.IMPORT) {
+      if (mode.isImportOrForwardMode()) {
```

Comment:
@lei-tianyu I presume negation is missing in this if(). This should be if (!mode.isImportOrForwardMode()) {